File: sales-frontend/src/api/customer.ts
import request from '@/utils/request'

export interface Customer {
  id?: number
  name?: string
  phone?: string
  company?: string
  position?: string
  industry?: string
  sourceType?: string
  content?: string
  requirement?: string
  status?: string
  salesId?: number
  attachmentCount?: number
  createTime?: string
}

// 添加客户
export const addCustomer = (data: FormData) => {
  return request.post<Customer>('/customer/add', data, {
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

// 查询客户列表
export const getCustomers = (salesId: number) => {
  return request.get<Customer[]>('/customer/list', {
    params: { salesId }
  })
}

// 查询客户详情
export const getCustomerDetail = (id: number) => {
  return request.get<Customer>(`/customer/${id}`)
}

// 更新客户
export const updateCustomer = (data: Customer) => {
  return request.put<boolean>('/customer/update', data)
}

// 删除客户
export const deleteCustomer = (id: number) => {
  return request.delete<boolean>(`/customer/${id}`)
}

File: backend/src/main/java/com/sales/customer/entity/Customer.java
package com.sales.customer.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("customer")
public class Customer {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String name;
    
    private String phone;
    
    private String company;
    
    private String position;
    
    private String industry;
    
    private String sourceType;
    
    private String content;
    
    private String imagePath;
    
    private String audioText;
    
    private String requirement;
    
    private String status;
    
    private Long salesId;
    
    private Integer attachmentCount;
    
    @TableField(fill = FieldFill.INSERT)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private LocalDateTime updateTime;
    
    @TableLogic
    private Integer deleted;
}
